### ISSUE TITLE
Changing wallpapers makes character look in random direction when game is run in gamescope

### ISSUE THREAD

**Issue by Roger-Roger-debug** [hyprwm/Hyprland#4497]
### Hyprland Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch HEAD at commit 934112af5bed36a1273cc0d5588e27ab4825922b dirty (config: Use canonical instead of read_symlink (4136)).
Tag: v0.33.1-34-g934112af

flags: (if any)

```

</details>

### Bug or Regression?

Regression

### Description

I have a script that calls `hyprctl hyprpaper` commands to change my wallpaper every few minutes. If I'm playing a game while the wallpaper changes my character will suddenly look in a random direction. This only happens if there was any input in between changes, if I just stand still and change the wallpaper multiple times the issue only occurs on the first change.

At first I thought I was also stumbling upon #3672 but a git bisect revealed the offending commit to be ba2af6f86d26fbf4b51aa37bf901e0118f46bbe3, which was introduced well after the original bug report.

Apologies if this is a gamescope bug and not related to hyprland.

### How to reproduce

1. Start a game in gamescope (tested Borderlands 2, Lethal Company)
2. get somewhere with first person view
3. Change your wallpaper
4. Observe that you are looking somewhere completely different

**Comment by Roger-Roger-debug** [comment]
This is from the logs when the bug occurs

```
[LOG] Set keyboard focus to surface 55f2410a0fe0, with [Window 55f2410eece0: title: "Borderlands 2 (32-bit, DX9)"]
[LOG] Searching for matching rules for gamescope (title: Borderlands 2 (32-bit, DX9))
[LOG] Window rule opaque -> gamescope matched [Window 55f2410eece0: title: "Borderlands 2 (32-bit, DX9)"]
[LOG] Window rule workspace 3 -> gamescope matched [Window 55f2410eece0: title: "Borderlands 2 (32-bit, DX9)"]
[LOG] Registered signal for owner 55f240cfe3e0: 55f2410a12b0 -> 55f240cfe438 (owner: Mouse constraint commit)
[LOG] Constrained mouse to 55f241ed94b0
[LOG] Keybind triggered, calling dispatcher (64, , 51)
[LOG] Callback 55f240cfe460 -> 55f240cfe458, Mouse constraint commit removed.
[LOG] Searching for matching rules for gamescope (title: Borderlands 2 (32-bit, DX9))
[LOG] Window rule opaque -> gamescope matched [Window 55f2410eece0: title: "Borderlands 2 (32-bit, DX9)"]
[LOG] Window rule workspace 3 -> gamescope matched [Window 55f2410eece0: title: "Borderlands 2 (32-bit, DX9)"]
```

Keybind event is me manually changing the wallpaper.

**Comment by Roger-Roger-debug** [comment]
Fixed by #4889
